crates/ott-common/src/discovery/dns.rs

```diff
@@ -29,15 +29,17 @@ fn deserialize_dns_server<'de, D>(deserializer: D) -> Result<Option<SocketAddr>,
 where
     D: Deserializer<'de>,
 {
-    let mut buf = String::deserialize(deserializer)?;
+    let buf = String::deserialize(deserializer)?;
 
-    if Option::is_none(&buf.find(':')) {
-        buf.push_str(":53");
+    if buf.is_empty() {
+        return Ok(None);
     }
 
-    Ok(Some(
-        SocketAddr::from_str(&buf).expect("Error: Could not unwrap"),
-    ))
+    if IpAddr::from_str(&buf).is_ok() {
+        Ok(Some(SocketAddr::new(IpAddr::from_str(&buf).unwrap(), 53)))
```

Comment:
`unwrap` panics, and you're repeating the `from_str` operation. use an `if let ...` or `match` instead